File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = false;
        
        Stack<Character> stack = new Stack<>();
        
        for (int i = 0; i < s.length(); i++) {
            Character str = s.charAt(i);
            if (str == '(') {
                stack.push(str);
            } else {
                if (stack.isEmpty()) {
                    return false; // 
                } else {
                    stack.pop();
                }
            }
        }
        
        if (stack.isEmpty()) {
            answer = true;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    stack = []
    
    for i in s:
        if not stack:
            stack.append(i)
        elif stack:
            if stack[-1] == "(" and i == ")":
                stack.pop()
            else:
                stack.append(i)
    
    if stack:
        return False
    return True